### ISSUE TITLE
duplicateCheck looks buggy

### ISSUE THREAD

**Issue by PalleZingmark** [rossPatton/stylint#27]
Hi,
It seems like *duplicateCheck* is a bit buggy.

I get a warning for the following code:
```css
.foo {
  .bar {
    color: red;
  }
}
```

> Warning:  duplicate property or selector, consider merging
> File: ./app/assets/stylus/fubar.styl
> Line: 5: }

Setting it to false has no effect either, same warning regardless of settings.
```javascript
...
"duplicates": false,
"globalDupe": false,
...
```

**Comment by rossPatton** [comment]
Hey @PalleZingmark thanks for pointing this out.

So, really we got 2 issues here. Duplicates can't toggle and the actual check itself isn't working as intended. 

Don't have a lot of time right now, so I just pushed up a quick fix for the toggle issue: if you update to the latest stylint you should be able to toggle duplicates off.

I'll take a longer look at false positives problem prolly tonight.

Thanks!

**Comment by PalleZingmark** [comment]
Thank you, toggling duplicates is now working as intended. :+1:

**Comment by rossPatton** [comment]
Hey, i wasn't able to replicate your issue, but I was able to make a few improvements to duplicates. I'd get the latest and see if they resolve your issue maybe?

I'm gonna close this issue out, i'll reopen if I can replicate the issue.

**Comment by PalleZingmark** [comment]
No, unfortunately it does not solve my issue.

I did some further tests, and while this throws a warning:
´´´css
.foo {
  .bar {
    color: red;
  }
}
´´´
This does not:
´´´css
.foo {
  .bar {
    color: red;
  }}
´´´

**Comment by rossPatton** [comment]
So, i still couldn't replicate this. Maybe post your config? Are you using tabs or spaces for indenting?

**Comment by PalleZingmark** [comment]
stylint-test.styl
```css
.foo {
  .foo-bar {
    color: red;
  }
}
```

.stylintrc
```json
{
    "alphabetical": false,
    "borderNone": true,
    "brackets": false,
    "colons": false,
    "commaSpace": true,
    "commentSpace": false,
    "cssLiteral": false,
    "duplicates": true,
    "depthLimit": 4,
    "efficient": true,
    "enforceVarStyle": false,
    "enforceBlockStyle": false,
    "extendPref": false,
    "globalDupe": false,
    "indentSpaces": 2,
    "leadingZero": false,
    "maxWarnings": 10,
    "mixed": true,
    "namingConvention": "BEM",
    "parenSpace": false,
    "placeholders": false,
    "semicolons": false,
    "Whitespace": true,
    "universal": false,
    "valid": true,
    "zeroUnits": true,
    "zIndexDuplicates": false,
    "zIndexNormalize": false
}
```

hmm, I can see some strange behavior. 

If I just lint **stylint-test.styl** by itself, using the config above - it passes.
But if I @import it into my project - it fails.
> Warning: duplicate property or selector, consider merging
> File: ./app/assets/stylus/stylint-test.styl
> Line: 5: }

I have no other classes named **.foo** or **.foo-bar** in my project, so it shouldn't throw a warning.

**Comment by rossPatton** [comment]
@PalleZingmark hey so, i was eventually able to replicate the issue and hopefully 0.8.8 fixes it.

Let me know!

**Comment by lin-hun** [comment]
.....in 0.8.8,i set duplicates as true.but didnt throw a warning
my test code:
```stylus
.some-class
  margin 0
  margin 5px

```

**Comment by rossPatton** [comment]
@lin-hun thx for reporting: was able to duplicate, try upgrading to 0.8.9 and see if that fixes it.

**Comment by lin-hun** [comment]
=w= duplicates works well

**Comment by rossPatton** [comment]
+1 cool, gonna close this now
